py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,400 @@
+                    amount=int(point[1]),
+                    usage_type=parsed_unit,
+                )
+            )
+
+    return record_list
+
+
+def post_to_usage_accumulator(
+    record_list: Sequence[UsageAccumulatorRecord],
+    usage_accumulator: UsageAccumulator,
+) -> None:
+    """
+    Posts UsageAccumulatorRecords to UsageAccumulator.
+    """
+    for record in record_list:
+        # TODO remove this before prod
+        # print(f"resource: {record.resource_id}, "
+        #       f"app_feature: {record.app_feature}, "
+        #       f"amount: {record.amount}, "
+        #       f"usage_type: {record.usage_type}")
+        usage_accumulator.record(
+            resource_id=record.resource_id,
+            app_feature=record.app_feature,
+            amount=record.amount,
+            usage_type=record.usage_type,
+        )
+
+
+def main(
+    query_file: TextIO,
+    start_time: int,
+    end_time: int,
+    kafka_config_file: TextIO,
+) -> None:
+    """
+    query_file: File with a list of dictionaries,
+                each containing a Datadog query and unit
+    start_time: Start of the time window in Unix epoch format
+                with second-level precision
+    end_time: End of the time window in Unix epoch format
+              with second-level precision
+    kafka_config_file: File with parameters to initialize
+                       UsageAccumulator object
+    """
+    kafka_config = parse_and_assert_kafka_config(kafka_config_file)
+    usage_accumulator = UsageAccumulator(kafka_config=kafka_config)
+
+    query_list = parse_and_assert_query_file(query_file)
+    record_list: List[UsageAccumulatorRecord] = []
+    for query_dict in query_list:
+        query, configured_unit = query_dict["query"], query_dict.get("unit")
+        assert_valid_query(query)
+
+        response = query_datadog(query, start_time, end_time)
+        series_list = parse_and_assert_response_series(response)
+
+        parsed_unit = parse_and_assert_unit(series_list)
+        if configured_unit:
+            warn_different_configured_and_dd_unit(configured_unit, parsed_unit)
+            unit = configured_unit
+        else:
+            unit = parsed_unit
+
+        assert_valid_unit(unit)
+        warn_multiple_units(series_list)
+        usage_unit = UsageUnit(unit.lower())
+
+        record_list.extend(process_series_data(series_list, usage_unit))
+
+    post_to_usage_accumulator(record_list, usage_accumulator)
+    usage_accumulator.flush()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="fetcher")
+
+    parser.add_argument(
+        "--query_file",
+        type=argparse.FileType("r"),
+        help="JSON file containing Datadog queries and units",
+    )
+    parser.add_argument(
+        "--start_time", type=int, help="Start of the query time window"
+    )
+    parser.add_argument(
+        "--end_time", type=int, help="End of the query time window"
+    )
```

Comment:
Please pass timesamp and period. 
if you pass start and end timestamp, in k8s you will have to calculate one from the other in the CLI. That is less robust than passing the period that is constant.